ydb/core/tx/schemeshard/ut_index_build/ut_vector_index_build.cpp

```diff
@@ -0,0 +1,245 @@
+            TableName: "Table"
+            IndexName: "index1"
+        )");
+        env.TestWaitNotification(runtime, txId, tenantSchemeShard);
+
+        RebootTablet(runtime, tenantSchemeShard, runtime.AllocateEdgeActor());
+
+        TestDescribeResult(DescribePath(runtime, tenantSchemeShard, "/MyRoot/ServerLessDB/Table"),
+                           {NLs::PathExist,
+                            NLs::IndexesCount(0),
+                            NLs::PathVersionEqual(8)});
+
+        // Test that index build succeeds on recreated columns
+
+        TestAlterTable(runtime, tenantSchemeShard, ++txId, "/MyRoot/ServerLessDB", R"(
+              Name: "Table"
+              DropColumns { Name: "embedding" }
+        )");
+        env.TestWaitNotification(runtime, txId, tenantSchemeShard);
+
+        TestAlterTable(runtime, tenantSchemeShard, ++txId, "/MyRoot/ServerLessDB", R"(
+              Name: "Table"
+              Columns { Name: "embedding"   Type: "String" }
+        )");
+        env.TestWaitNotification(runtime, txId, tenantSchemeShard);
+
+        TestBuildVectorIndex(runtime, ++txId, tenantSchemeShard, "/MyRoot/ServerLessDB", "/MyRoot/ServerLessDB/Table", "index2", "embedding");
+        env.TestWaitNotification(runtime, txId, tenantSchemeShard);
+
+        // CommonDB
+        TestCreateExtSubDomain(runtime, ++txId,  "/MyRoot",
+                               "Name: \"CommonDB\"");
+        env.TestWaitNotification(runtime, txId);
+
+        TestAlterExtSubDomain(runtime, ++txId,  "/MyRoot",
+                              "StoragePools { "
+                              "  Name: \"pool-3\" "
+                              "  Kind: \"pool-kind-3\" "
+                              "} "
+                              "PlanResolution: 50 "
+                              "Coordinators: 1 "
+                              "Mediators: 1 "
+                              "TimeCastBucketsPerMediator: 2 "
+                              "ExternalSchemeShard: true "
+                              "Name: \"CommonDB\"");
+        env.TestWaitNotification(runtime, txId);
+
+        TestDescribeResult(DescribePath(runtime, "/MyRoot/CommonDB"),
+                           {NLs::PathExist,
+                            NLs::IsExternalSubDomain("CommonDB"),
+                            NLs::ExtractTenantSchemeshard(&tenantSchemeShard)});
+
+        TestCreateTable(runtime, tenantSchemeShard, ++txId, "/MyRoot/CommonDB", R"(
+              Name: "Table"
+              Columns { Name: "key"       Type: "Uint32" }
+              Columns { Name: "embedding" Type: "String" }
+              KeyColumnNames: ["key"]
+        )");
+        env.TestWaitNotification(runtime, txId, tenantSchemeShard);
+
+        for (ui32 key = 100; key < 300; ++key) {
+            fnWriteRow(TTestTxConfig::FakeHiveTablets + 6, key, std::to_string(key), "Table");
+        }
+
+        TVector<TString> billRecords;
+        runtime.SetObserverFunc([&](TAutoPtr<IEventHandle>& ev) {
```

Comment:
SetObserver is [deprecated](https://github.com/ydb-platform/ydb/blob/5ccb8183a3926fca9bb3bdf96408c9314102cef1/ydb/library/actors/testlib/test_runtime.h#L239C74-L239C101), use [AddObserver](https://github.com/ydb-platform/ydb/blob/main/ydb/library/actors/testlib/test_runtime.h#L362)